example/src/main/java/eu/webeid/example/security/ui/WebEidLoginPageGeneratingFilter.java

```diff
@@ -0,0 +1,132 @@
+        <body>
+        <div id="error-message" class="alert alert-danger" style="display: none;" role="alert">
+          <div class="message"></div>
+          <pre class="details"></pre>
+        </div>
+
+        <script type="module">
+        import { showErrorMessage } from "/js/errors.js";
+        (function () {
+          const frag = location.hash ? location.hash.substring(1) : "";
+          if (!frag) { showErrorMessage({ code: "UNKNOWN_ERROR", message: "Missing authentication payload" }); return; }
+
+          let payload;
+          try { payload = JSON.parse(atob(frag)); } catch (e) {
+            console.error("Failed to parse payload", e);
+            showErrorMessage({ code: "UNKNOWN_ERROR", message: "Failed to parse authentication payload" });
+            return;
+          }
+
+          if (payload["error"]) {
+            showErrorMessage({
+              code: payload["code"] ?? "UNKNOWN_ERROR",
+              message: payload["message"] ?? "Authentication failed"
+            });
+            return;
+          }
+
+          const authToken = payload["auth-token"];
+
+          fetch("%s", {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+              "%s": "%s"
+            },
+            body: JSON.stringify(authToken),
+            credentials: "include"
+          })
+          .then(r => {
+            if (!r.ok) throw new Error("HTTP " + r.status);
+            window.location.replace("/welcome");
+          })
+          .catch(error => {
+            showErrorMessage(error);
+          });
+        })();
+        </script>
+        </body>
+        </html>
+        """;
+    private final RequestMatcher requestMatcher;
+    private final String loginProcessingPath;
+
+    public WebEidLoginPageGeneratingFilter(String path, String loginProcessingPath) {
+        this.requestMatcher = PathPatternRequestMatcher.withDefaults().matcher(HttpMethod.GET, path);
+        this.loginProcessingPath = loginProcessingPath;
+    }
+
+    @Override
+    protected void doFilterInternal(@NonNull HttpServletRequest request, @NonNull HttpServletResponse response, @NonNull FilterChain chain)
+        throws IOException, ServletException {
+        if (!requestMatcher.matches(request)) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        var csrf = (CsrfToken) request.getAttribute(CsrfToken.class.getName());
+        if (csrf == null) {
+            csrf = (CsrfToken) request.getAttribute("_csrf");
+        }
+
+        String html = generateHtml(csrf);
+
+        response.setContentType(MediaType.TEXT_HTML_VALUE);
+        response.getWriter().write(html);
+    }
+
+    private String generateHtml(CsrfToken csrf) {
+        return String.format(
+            LOGIN_PAGE_HTML,
+            loginProcessingPath,
+            csrf != null ? csrf.getHeaderName() : "X-CSRF-TOKEN",
+            csrf != null ? csrf.getToken() : ""
```

Comment:
```suggestion
    private String renderTemplate(HttpServletRequest request, HttpServletResponse response, CsrfToken csrf) {
        IWebExchange exchange = webApp.buildExchange(request, response);
        var locale = RequestContextUtils.getLocale(request);
        var ctx = new WebContext(exchange, locale);
        ctx.setVariable("loginProcessingPath", loginProcessingPath);
        ctx.setVariable("csrfHeaderName", csrf != null ? csrf.getHeaderName() : "X-CSRF-TOKEN");
        ctx.setVariable("csrfToken", csrf != null ? csrf.getToken() : "");
        return templateEngine.process("webeid-login", ctx);
```

and add `import org.springframework.web.servlet.support.RequestContextUtils`.